source/partial-on-undefined-deep.d.ts

```diff
@@ -0,0 +1,72 @@
+import type {BuiltIns} from './internal';
+import type {Merge} from './merge';
+
+/**
+@see PartialOnUndefinedDeep
+*/
+export interface PartialOnUndefinedDeepOptions {
+	/**
+	Whether to affect the individual elements of arrays and tuples
+
+	@default false
+	*/
+	readonly recurseIntoArrays?: boolean;
+}
+
+/**
+Create a deep version of another type where all keys accepting `undefined` type are set to optional.
+
+This utility type is recursive, transforming at any level deep.
+By default, this type does not affects array and tuple types unless you explicitly pass `{recurseIntoArrays: true}` as the second type argument.
```

Comment:
It's confusing to say that it would "affect" arrays, as it's for objects. It's more correct to say that the option makes it affect items of arrays/tuples.